### ISSUE TITLE
error: date/time field value out of range: "0000-00-00 00:00:00" for postgres

### ISSUE THREAD

**Issue by tw0517tw** [bsiddiqui/bookshelf-paranoia#36]
postgreSQL can't use ALL_ZERO datetime to query.

failed knex query is like:
```js
{ method: 'select',
  options: {},
  timeout: false,
  cancelOnTimeout: false,
  bindings: [ '5afa50be3a3ff4506727e85e', '0000-00-00 00:00:00', 1 ],
  __knexQueryUid: 'e3f9ee71-38e4-426c-a593-38777954a348',
  sql: 'select "admins".* from "admins" where "admins"."some_id" = ? and "admins"."deleted_at" is null or "admins"."deleted_at" = ? limit ?' }
```

```
error: date/time field value out of range: "0000-00-00 00:00:00"
  at Connection.parseE (/app/node_modules/pg/lib/connection.js:553:11)
  at Connection.parseMessage (/app/node_modules/pg/lib/connection.js:378:19)
  at Socket.<anonymous> (/app/node_modules/pg/lib/connection.js:119:22)
  at emitOne (events.js:116:13)
  at Socket.emit (events.js:211:7)
  at addChunk (_stream_readable.js:263:12)
  at readableAddChunk (_stream_readable.js:250:11)
  at Socket.Readable.push (_stream_readable.js:208:10)
  at TCP.onread (net.js:607:20)
```

**Comment by bsiddiqui** [comment]
@tw0517tw I believe the most recent patch fixes this issue

**Comment by tw0517tw** [comment]
@bsiddiqui Sorry but I've tried `v0.12.1` and it didn't fix this. It seems we shouldn't put `"0000-00-00 00:00:00"` into any part of postgres date/time queries.

**Comment by bsiddiqui** [comment]
@tw0517tw that's interesting, it's just a string. Do you know why that would be?

**Comment by tw0517tw** [comment]
We create our `deleted_at` column with [`timestamp()`](https://knexjs.org/#Schema-timestamp) type, so postgres is doing validation and `"0000-00-00 00:00:00"` fails the validation as timestamp starts at 1970/1/1.

**Comment by bsiddiqui** [comment]
Alright I'll push a fix tomorrow

**Comment by tw0517tw** [comment]
Thanks so much!

**Comment by bsiddiqui** [comment]
@tw0517tw fixed in `0.13.0`
